src/refiners/foundationals/latent_diffusion/stable_diffusion_1/unet.py

```diff
@@ -280,10 +280,13 @@ def init_context(self) -> Contexts:
             "diffusion": {"timestep": None},
             "range_adapter": {"timestep_embedding": None},
             "sampling": {"shapes": []},
+            "ip_mask": {"mask": None},
         }
 
     def set_clip_text_embedding(self, clip_text_embedding: Tensor) -> None:
         self.set_context("cross_attention_block", {"clip_text_embedding": clip_text_embedding})
 
     def set_timestep(self, timestep: Tensor) -> None:
         self.set_context("diffusion", {"timestep": timestep})
+    def set_ip_mask(self, mask: Tensor) -> None:
```

Comment:
I think this method should belong to IPAdapter. One of the core design patterns of Refiners is to try to modify the previous code as little as possible. In this case, unet.py should know very little about IPAdapter.

Reply:
Thanks for the check! Ah let me learn more about the context API haha

Reply:
We should probably write a notebook to clarify how the Context API is working 😅

Reply:
Ok. I think I did this